src/main/java/org/codeforamerica/messaging/providers/mailgun/MailgunSignatureVerificationService.java

```diff
@@ -0,0 +1,44 @@
+package org.codeforamerica.messaging.providers.mailgun;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import jakarta.annotation.PostConstruct;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.math.BigInteger;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+@Service
+@Slf4j
+public class MailgunSignatureVerificationService {
+    @Value("${mailgun.webhook.signing.key}")
+    private String mailgunWebhookSigningKey;
+
+    Mac mac;
+
+    @PostConstruct
+    public void initializeMac() {
+        SecretKeySpec secretKeySpec = new SecretKeySpec(mailgunWebhookSigningKey.getBytes(), "HmacSHA256");
+        try {
+            this.mac = Mac.getInstance("HmacSHA256");
+            mac.init(secretKeySpec);
```

Comment:
This is new to me, but completely makes sense to do it this way now that I can see how to do it.

Reply:
This particular code fragment is only needed for `Mailgun`. `Twilio` is easier as they provide `RequestValidator`.

Reply:
I guess I was thinking this could be used in the unit tests to simulate the RequestValidator with a fake signature, but even though I can see how it could be done, I'm not sure if it is worth doing for testing purposes